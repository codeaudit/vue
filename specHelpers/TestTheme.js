import {setOpacity, toRgbaString} from '@andrew-codes/color-functions';

export default {
    _name: 'Test Theme',
    altColor: '#ffa500',
    basicColor: '#00ffff',
    basicFontFamily: '\'Proxima Nova\', \'Lucida Sans Unicode\', \'Lucida Grande\', sans-serif',
    baseIconSize: 24,
    boldFont: 600,
    borderPrimaryColor: '#000',
    darkInverseColor: '#fff',
    disabledPrimaryColor: '#808080',
    errorPrimaryColor: '#ffc0cb',
    errorSecondaryColor: '#ffc0cd',
    fieldBorderColor: '#808080',
    focusedPrimaryColor: '#00ffff',
    focusedSecondaryColor: toRgbaString(setOpacity('#00ffff', 0.5)),
    gutter: 8,
    importantColor: '#f99643',
    largeFontSize: 22,
    largeGutter: 12,
    largeLineHeight: 2.285,
    lightInverseColor: '#030303',
    mediumFontSize: 16,
    normalBackground: '#fff',
    normalLineHeight: 1.285,
    normalRadius: 3,
    requiredPrimaryColor: '#ffc0cb',
    pendingPrimaryColor: '#ffff00',
    smallFontSize: 14,
    smallGutter: 6,
    textPrimaryColor: '#030303',
    textSecondaryColor: '#030303',
    textDisabledColor: '#808080',
    xLargeFontSize: 24,
    xLargeGutter: 16,
    xMediumFontSize: 18,
    xSmallFontSize: 8,
    xxSmallGutter: 3
};
