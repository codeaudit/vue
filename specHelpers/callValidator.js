import ReactPropTypesArgs from './reactPropTypeArgs';

const callValidator = (validator, element, propName = '', componentName = '') => validator(element.props, propName, componentName, ...ReactPropTypesArgs);

export const validatorPasses = (validator, element, propName) => callValidator(validator, element, propName) === null;
export const validatorFails = (validator, element, propName) => callValidator(validator, element, propName) instanceof Error;
