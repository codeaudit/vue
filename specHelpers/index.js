import React from 'react';
import ThemeProvider, {getTheme} from './../src/Theme';

export const applyTheme = (ComponentUnderTest, theme = getTheme()) => <ThemeProvider theme={theme}>{ComponentUnderTest}</ThemeProvider>;
