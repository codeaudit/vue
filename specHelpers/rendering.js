import toJson from 'enzyme-to-json';
import {mount, shallow} from 'enzyme';
import React from 'react';
import theme from './TestTheme';

export const getRender = (Component) => (props = {}, context = {theme}) => shallow(<Component {...props} />, {context});
export const getMount = (Component) => (props = {}, context = {theme}) => mount(<Component {...props} />, {context});
export const snapshot = (renderedComponent) => toJson(renderedComponent);
