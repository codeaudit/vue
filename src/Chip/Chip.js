import React, {PropTypes} from 'react';
import {darken, toRgbaString} from '@andrew-codes/color-functions';
import Radium from './../utilities/Radium';
import ThemeProvider from './../Theme';
import {createEventHandler} from './../utilities/component';
import {IconButton} from './../Button';
import {CloseIcon} from './../Icons';
import {xxSmall} from './../Button/Sizes';

const getStyles = (props, context) => {
    const {
        backgroundColor,
        color,
        fontSize,
        fullWidth,
        lineHeight,
        width,
    } = props;
    const {
        basicFontFamily,
        normalRadius,
        smallGutter,
        xxSmallGutter,
    } = context.theme;

    return ({
        buttonWrapper: {
            display: 'flex',
            marginLeft: `${fontSize}px`,
        },
        root: {
            backgroundColor,
            borderRadius: `${normalRadius}px`,
            boxSizing: 'border-box',
            color,
            display: 'flex',
            fontSize: `${fontSize}px`,
            lineHeight: `${lineHeight}`,
            margin: `${xxSmallGutter}px`,
            padding: `${smallGutter}px`,
            width: fullWidth ? '100%' : `${width}px`,
        },
        text: {
            alignSelf: 'center',
            flex: 1,
            fontFamily: basicFontFamily,
        },
        wrapper: {
            alignSelf: 'center',
            display: 'flex',
            width: '100%',
        },
    });
};

const Chip = (props, context) => {
    const {
        backgroundColor,
        color,
        fontSize,
        fullWidth,
        lineHeight,
        oid,
        onRequestRemove,
        text,
        width,
    } = props;
    const styles = getStyles({
        backgroundColor,
        color,
        fontSize,
        fullWidth,
        width,
    }, context);
    const iconButtonDarkenCoefficient = 0.45;
    const handleClick = createEventHandler(onRequestRemove, oid);

    return (
        <div style={styles.root}>
            <div style={styles.wrapper}>
                <span style={styles.text}>{text}</span>
                <div style={styles.buttonWrapper}>
                    <IconButton
                        circle
                        color={toRgbaString(darken(backgroundColor, iconButtonDarkenCoefficient))}
                        icon={CloseIcon}
                        size={xxSmall}
                        width={fontSize * lineHeight}
                        onClick={handleClick}
                    />
                </div>
            </div>
        </div>
    );
};
Chip.propTypes = {
    /**
     * Background color of the chip
     */
    backgroundColor: PropTypes.string,
    /**
     * Font color of the chip
     */
    color: PropTypes.string,
    /**
     * Font size used for the text of the Chip
     */
    fontSize: PropTypes.number,
    /**
     * The Chip will expand to fill the container's space if true; otherwise it will inline-block like with a width
     */
    fullWidth: PropTypes.bool,
    /**
     * Line height for the text of the Chip
     */
    lineHeight: PropTypes.number,
    /**
     * Unique identifier for this Chip
     */
    oid: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
    ]),
    /**
     * Function handling the clicking of the close icon of the Chip
     */
    onRequestRemove: PropTypes.func,
    /**
     * The text to render on the Chip
     */
    text: PropTypes.string.isRequired,
    /**
     * Width of the Chip
     */
    width: PropTypes.number,
};
Chip.defaultProps = {
    backgroundColor: '#ccc',
    color: '#000',
    fontSize: 14,
    fullWidth: false,
    lineHeight: 1,
    onRequestRemove: () => {
    },
    width: 200,
};
Chip.contextTypes = {
    theme: PropTypes.shape(ThemeProvider.themeDefinition).isRequired,
};
Chip.displayName = 'Chip';
export default Radium(Chip);
