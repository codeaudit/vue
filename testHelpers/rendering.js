import toJson from 'enzyme-to-json';
import React, {PropTypes} from 'react';
import {mount, render, shallow} from 'enzyme';
import theme from './../packages/themes/src/V1Theme';

const ctx = {
    theme,
};

export const getShallow = (Component) => (props = {}, context = {}) => shallow(<Component {...props} />, {
    childContextTypes: {
        theme: PropTypes.object,
    },
    context: {
        ...ctx,
        ...context,
    },
});

export const getRender = (Component) => (props = {}, context = {}) => render(<Component {...props} />, {
    childContextTypes: {
        theme: PropTypes.object,
    },
    context: {
        ...ctx,
        ...context,
    },
});

export const getMount = (Component) => (props = {}, options = {}, context = {}) => mount(<Component {...props} />, {
    childContextTypes: {
        theme: PropTypes.object,
    },
    context: {
        ...ctx,
        ...context,
    },
    ...options,
});

const replaceUIKey = (componentName) => {
    let index = 0;
    return (json) => {
        if (!json.props || !json.props.uiKey) {
            return json;
        }
        const uiKey = `${componentName}-${index}`;
        index += 1;
        return Object.assign({}, json, {
            uiKey,
            uiPath: [uiKey],
        });
    };
};
const replaceUiKeyRecursively = (json) => {
    if (!json.type) {
        return json;
    }
    const replaceUIKeyInParent = replaceUIKey(json.type);
    const output = replaceUIKeyInParent(json);
    if (!output.children) {
        return json;
    }
    output.children = json.children.map((child, index) => replaceUiKeyRecursively(child));
    return output;
};

export const snapshot = (renderedComponent) => {
    return toJson(renderedComponent);
};

export const reset = (component) => () => {
    if (component) {
        component.unmount();
    }
    document.body.innerHTML = '';
};
