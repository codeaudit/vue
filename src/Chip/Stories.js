import React from 'react';
import {action, storiesOf} from '@kadira/storybook';
import Chip from './';

storiesOf('Chip')
    .addWithInfo('defaults',
        ``,
        () => (
            <Chip
                text="Hello world"
                onRequestRemove={action('request removal')}
            />
        )
    )
    .addWithInfo('full width',
        ``,
        () => (
            <Chip
                fullWidth
                text="Hello world"
                onRequestRemove={action('request removal')}
            />
        )
    )
    .addWithInfo('width of 350px',
        ``,
        () => (
            <Chip
                text="Hello world"
                width={350}
                onRequestRemove={action('request removal')}
            />

        )
    );
