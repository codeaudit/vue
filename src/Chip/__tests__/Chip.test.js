import Chip from './../Chip';
import {getShallow, snapshot} from './../../../specHelpers/rendering';

const shallowRenderChip = getShallow(Chip);
const clickEvent = {test: true};

test('it can render text in the Chip', () => {
    const component = shallowRenderChip({
        text: 'Hello Chip',
    });
    expect(snapshot(component)).toMatchSnapshot();
});

test('it can render as full width', () => {
    const component = shallowRenderChip({
        fullWidth: true,
        text: 'Hello Chip',
    });
    expect(snapshot(component)).toMatchSnapshot();
});

test('it is click-able', () => {
    const onRequestRemove = jest.fn();
    const component = shallowRenderChip({
        oid: 'Chip:1',
        onRequestRemove,
        text: 'Hello Chip',
    });
    simulateRemoveChipClick(component, clickEvent);
    expect(onRequestRemove).toHaveBeenCalledTimes(1);
    expect(onRequestRemove).toHaveBeenCalledWith(clickEvent, 'Chip:1');
});

function simulateRemoveChipClick(wrapper, evt = {}) {
    wrapper.find('IconButton').simulate('click', evt);
}
