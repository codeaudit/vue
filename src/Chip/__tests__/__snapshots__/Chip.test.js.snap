exports[`test it can render as full width 1`] = `
<div
  data-radium={true}
  style={
    Object {
      "backgroundColor": "#ccc",
      "borderRadius": "3px",
      "boxSizing": "border-box",
      "color": "#000",
      "display": "flex",
      "fontSize": "14px",
      "lineHeight": "undefined",
      "margin": "3px",
      "padding": "6px",
      "width": "100%",
    }
  }>
  <div
    data-radium={true}
    style={
      Object {
        "alignSelf": "center",
        "display": "flex",
        "width": "100%",
      }
    }>
    <span
      data-radium={true}
      style={
        Object {
          "alignSelf": "center",
          "flex": 1,
          "fontFamily": "\'Proxima Nova\', \'Lucida Sans Unicode\', \'Lucida Grande\', sans-serif",
        }
      }>
      Hello Chip
    </span>
    <div
      data-radium={true}
      style={
        Object {
          "display": "flex",
          "marginLeft": "14px",
        }
      }>
      <IconButton
        backgroundColor="rgba(0,0,0,0)"
        circle={true}
        color="rgba(89,89,89,1)"
        disabled={false}
        icon={[Function]}
        onClick={[Function]}
        size={0.375}
        transition="all 0.25s linear 0ms"
        width={14} />
    </div>
  </div>
</div>
`;

exports[`test it can render text in the Chip 1`] = `
<div
  data-radium={true}
  style={
    Object {
      "backgroundColor": "#ccc",
      "borderRadius": "3px",
      "boxSizing": "border-box",
      "color": "#000",
      "display": "flex",
      "fontSize": "14px",
      "lineHeight": "undefined",
      "margin": "3px",
      "padding": "6px",
      "width": "200px",
    }
  }>
  <div
    data-radium={true}
    style={
      Object {
        "alignSelf": "center",
        "display": "flex",
        "width": "100%",
      }
    }>
    <span
      data-radium={true}
      style={
        Object {
          "alignSelf": "center",
          "flex": 1,
          "fontFamily": "\'Proxima Nova\', \'Lucida Sans Unicode\', \'Lucida Grande\', sans-serif",
        }
      }>
      Hello Chip
    </span>
    <div
      data-radium={true}
      style={
        Object {
          "display": "flex",
          "marginLeft": "14px",
        }
      }>
      <IconButton
        backgroundColor="rgba(0,0,0,0)"
        circle={true}
        color="rgba(89,89,89,1)"
        disabled={false}
        icon={[Function]}
        onClick={[Function]}
        size={0.375}
        transition="all 0.25s linear 0ms"
        width={14} />
    </div>
  </div>
</div>
`;
